met-api/src/met_api/models/survey.py

```diff
@@ -89,3 +96,17 @@ def update_survey(cls, survey: SurveySchema) -> DefaultMethodResult:
         query.update(update_fields)
         db.session.commit()
         return DefaultMethodResult(True, 'Survey Updated', survey_id)
+
+    @classmethod
+    def link_survey(cls, survey_id, engagement_id) -> DefaultMethodResult:
+        """Update engagement."""
+        update_fields = dict(
+            engagement_id=engagement_id,
+        )
+        query = Survey.query.filter_by(id=survey_id)
+        record = query.first()
+        if not record:
+            return DefaultMethodResult(False, 'Survey Not Found', survey_id)
+        query.update(update_fields)
+        db.session.commit()
```

Comment:
I think if you just do survey.engagement_id = eng_id and session.commit ;it will all work...might increase the readability ..

Reply:
noted will do